encoding/value/value.rs

```diff
@@ -458,7 +458,13 @@ impl fmt::Display for Value<'_> {
         match self {
             Value::Boolean(bool) => write!(f, "{bool}"),
             Value::Integer(integer) => write!(f, "{integer}"),
-            Value::Double(double) => write!(f, "{double}"),
+            Value::Double(double) => {
+                if double.fract() == 0.0 {
+                    write!(f, "{double}.0")
+                } else {
+                    write!(f, "{double}")
```

Comment:
Our doubles in typeql always must contain a decimal value to be re-parse-able. Technically we should probably implement `TypeQLSyntax` instead of the default `Display` being TypeQL compatible, but I think this is fine (we already were doing this a bit with the Decimal display implementation anyway, which included the required `dec` suffix!)